excavator/utils.py

```diff
@@ -78,8 +83,13 @@ def env_int(name, required=False, default=empty):
     environment, a `TypeError` will be raised since we don't want to guess
     about a sensible default.
     """
-    value = get_env_value(name, required=required, default=default)
+    if required and default is not empty:
+        raise ValueError("Using `default` with `required=True` is invalid")
+
+    value = get_env_value(name, required=required, default=empty)
     if value is empty:
+        if default is not empty:
+            return default
```

Comment:
No need to nest this (`python -m this | tail -n 15 | head -n 1`), when you could just write:

```python
if value is empty and default is not empty:
    return default
```